jbehave-support-core/src/main/java/org/jbehavesupport/core/internal/web/action/SelectWebAction.java

```diff
@@ -23,9 +23,17 @@ public void perform(WebActionContext ctx) {
 
         WebElement element = findElement(ctx);
 
-        Assert.isTrue(equalsIgnoreCase(element.getTagName(), "select"), "The SELECT action is not supported on this element type: " + element.getTagName());
-
-        new Select(element).selectByVisibleText(ctx.getData());
+        if (equalsIgnoreCase(element.getTagName(), "input") && equalsIgnoreCase(element.getAttribute("type"), "checkbox")){
+            boolean checked = ctx.getData().isEmpty() || toBoolean(ctx.getData());
+
+            if (toBoolean(element.getAttribute("checked")) ^ checked){
+                element.sendKeys(Keys.SPACE);
+            }
+        } else if (equalsIgnoreCase(element.getTagName(), "select")){
+            new Select(element).selectByVisibleText(ctx.getData());
+        } else {
+            throw new IllegalArgumentException("The SELECT action is not supported on this element type: " + element.getTagName());
+        }
```

Comment:
minor: I'd try to decrease the complexity of this method by extracting into smaller and more descriptive methods ... what about something like this?

```
if( isCheckbox(element) ) {
    handleCheckbox(element, ctx);
} else if( isSelect(element) ) {
    handleSelect(element, ctx);
} else {
```